doc/build-helpers/trivial-build-helpers.chapter.md

```diff
@@ -84,37 +175,265 @@ writeTextFile {
+
+`text`
+
+: the contents of the file.
+
+The created file is marked as executable.
+
+The file's contents will be put into `/nix/store/<store path>/bin/<name>`.
+
+The store path will include the the name, and it will be a directory. Any path separators and shell-reserved elements in the name are escaped to create the store path identifier.
+
+::: {.example #ex-writeScriptBin}
+# Usage of `writeScriptBin`
+```nix
+writeScriptBin "my-script"
   ''
-  Contents of File
+  echo "hi"
+  '';
+```
+:::
+
+The example is a simpler way to spell:
+
+```nix
+writeTextFile {
+  name = "my-script";
+  text = ''
+    echo "hi"
   '';
-# Writes my-file to /nix/store/<store path> and makes executable.
-writeShellScript "my-file"
+  executable = true;
+  destination = "bin/my-script"
+}
+```
+
+### `writeShellScript` {#trivial-builder-writeShellScript}
+
+Writes a shell script to the store.
+
+`writeShellScript` takes two arguments: `name` and `text`, each of which should be a string.
+
+`name`
+
+: the name used in the Nix store path.
+
+`text`
+
+: the contents of the file.
+
+The created file is marked as executable.
+
+This function is almost exactly like `writeScript`, but it prepends a shebang line that points to the runtime shell (usually bash) at the top of the file contents.
+
+The store path will include the the name, and it will be a file. Any path separators and shell-reserved elements in the name are escaped to create the store path identifier.
+
+Here is an example.
+
+::: {.example #ex-writeShellScript}
+# Usage of `writeShellScript`
+```nix
+writeShellScript "my-script"
   ''
-  Contents of File
+  echo "hi"
+  '';
+```
+:::
+
+The example is a simpler way to spell:
+
+```nix
+writeTextFile {
+  name = "my-script";
+  text = ''
+    #! ${pkgs.runtimeShell}
+    echo "hi"
   '';
-# Writes my-file to /nix/store/<store path>/bin/my-file and makes executable.
-writeShellScriptBin "my-file"
+  executable = true;
+}
+```
+### `writeShellScriptBin` {#trivial-builder-writeShellScriptBin}
+
+Writes a shell script to a "bin" subdirectory of subdirectory of the store.
+
+`writeShellScriptBin` takes two arguments: `name` and `text`, each of which should be a string.
+
+`name`
+
+: the name used in the Nix store path and within the file generated under the store path.
+
+`text`
+
+: the contents of the file.
+
+This function is almost exactly like `writeScriptBin`, but it prepends a shebang line that points to the runtime shell (usually bash) at the top of the file contents.
+
+The file's contents will be put into `/nix/store/<store path>/bin/<name>`.
+
+The store path will include the the name, and it will be a directory. Any path separators and shell-reserved elements in the name are escaped to create the store path identifier.
+
+::: {.example #ex-writeShellScriptBin}
+# Usage of `writeShellScriptBin`
+```nix
+writeShellScriptBin "my-script"
   ''
-  Contents of File
+  echo "hi"
   '';
+```
+:::
+
+The example is a simpler way to spell:
```

Comment:
```suggestion
This example is equivalent to:
```

(Idea: avoiding colloquialisms, for those who might not have English as a culturally-familiar language.)

(Provided as a suggestion only in case you find it okay to apply as is.)